Core/Goals/CombatGoal.cs

```diff
@@ -39,7 +39,7 @@ public CombatGoal(ILogger logger, ConfigurableInput input, PlayerReader playerRe
             AddPrecondition(GoapKey.incombat, true);
             AddPrecondition(GoapKey.hastarget, true);
             AddPrecondition(GoapKey.targetisalive, true);
-            AddPrecondition(GoapKey.incombatrange, true);
+            //AddPrecondition(GoapKey.incombatrange, true);
```

Comment:
The idea behind the `incombatrange` you can only use any spells if you are in range, otherwise you have to approach the target. Which is the `ApproachTargetGoal` responsibility.

Reply:
My main idea to make bot to use approach time by time in the combat routine because incombatrange didn't work properly.